cuda/solver/lower_trs_kernels.cu

```diff
@@ -73,9 +73,10 @@ template <typename ValueType, typename IndexType>
 void generate(std::shared_ptr<const CudaExecutor> exec,
               const matrix::Csr<ValueType, IndexType>* matrix,
               std::shared_ptr<solver::SolveStruct>& solve_struct,
-              bool unit_diag, const gko::size_type num_rhs)
+              bool unit_diag, const solver::trisolve_algorithm algorithm,
+              const gko::size_type num_rhs)
 {
-    if (matrix->get_strategy()->get_name() == "sparselib") {
+    if (algorithm == solver::trisolve_algorithm::sparselib) {
```

Comment:
What happens if `trisolve_algorithm::ginkgo` is chosen? I did not see any branch for that option. It it actually not doing anything?

Reply:
Looks like you don't have / need a generation step in that case, so I will mark it as resolved.